### ISSUE TITLE
[BUG] Can't use sklearn transforms in 2.2.0 docker image

### ISSUE THREAD

**Issue by d-a-bunin** [etna-team/etna#92]
### 🐛 Bug Report

The error:
```
AttributeError: module 'numpy' has no attribute 'warnings'
```

As I understand the problem is the conflict of versions between sklearn and numpy. We should add a working restriction on the versions of the libraries.

The easy ad-hoc solution is to install `numpy<1.24`.

### Expected behavior

There should be no error.

I think we should also make sure to better check our docker images. Probable solutions:
- Test docker images,
- Build images on versions from `poetry.lock` to make sure that it works fine.

### How To Reproduce

Use docker image: `ghcr.io/tinkoff-ai/etna/etna-cpu:2.2.0`

Run script:
```python
import sys
from typing import Any
from typing import Sequence
import json

import warnings

warnings.filterwarnings("ignore")


def recursive_set_value(d: dict, key: Any, value: Any) -> None:
    if key in d:
        d[key] = value
    else:
        for _, v in d.items():
            if isinstance(v, dict):
                recursive_set_value(v, key, value)
            if isinstance(v, Sequence):
                for x in v:
                    if isinstance(x, dict):
                        recursive_set_value(x, key, value)


def run(cfg: dict) -> None:
    from etna.commands import mult
    from etna.datasets import TSDataset
    from etna.datasets.datasets_generation import generate_from_patterns_df
    from etna.metrics import MAE
    from etna.metrics import MSE
    from etna.metrics import SMAPE
    from etna.metrics import MedAE
    from etna.metrics import Sign
    from etna.pipeline import Pipeline
    from hydra_slayer import get_from_params
    from omegaconf import OmegaConf

    periods_x_freq = {
        "D": 300,
        "H": 300 * 24,
        "T": 10 * 24 * 60,
        "MS": 50,
        "W-MON": 100,
        "W-SUN": 100,
        "W": 100,
    }

    horizon_x_freq = {
        "D": 7,
        "H": 7,
        "T": 7,
        "MS": 7,
        "W-MON": 2,
        "W-SUN": 2,
        "W": 2,
    }

    freq = cfg["freq"]
    forecaster = cfg["forecaster"]

    ts = generate_from_patterns_df(
        periods=periods_x_freq[freq],
        start_time="1990-01-01",
        freq=freq,
        patterns=[
            [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
            [100, 90, 80, 70, 60, 50, 40, 30, 20, 10],
            [20, 40, 40, 50],
        ],
    )

    ts = TSDataset.to_dataset(ts)
    ts = TSDataset(ts, freq=freq)

    # config
    recursive_set_value(forecaster, "horizon", horizon_x_freq[freq])
    config = OmegaConf.create(forecaster)
    config = OmegaConf.to_container(config, resolve=True)

    # pipeline
    metrics = [Sign(), SMAPE(), MAE(), MSE(), MedAE()]
    pipeline: Pipeline = get_from_params(**config)
    metrics_df, forecast_df, fold_info_df = pipeline.backtest(
        ts,
        metrics=metrics,
        **{
            "n_folds": 1,
            "n_jobs": 1,
        },
    )

    print(metrics_df)


if __name__ == "__main__":
    cfg_str = '{"forecaster": {"_target_": "etna.pipeline.Pipeline", "horizon": 14, "model": {"_target_": "etna.models.ProphetModel"}, "transforms": [{"_target_": "etna.transforms.TimeSeriesImputerTransform", "in_column": "target", "strategy": "forward_fill", "window": -1}, {"_target_": "etna.transforms.MedianOutliersTransform", "in_column": "target"}, {"_target_": "etna.transforms.TimeSeriesImputerTransform", "in_column": "target", "strategy": "mean"}, {"_target_": "etna.transforms.YeoJohnsonTransform", "in_column": "target"}, {"_target_": "etna.transforms.LinearTrendTransform", "in_column": "target"}]}, "freq": "D", "tasks": ["forecast"], "version": "2.0.0"}'
    cfg = json.loads(cfg_str)

    run(cfg)

```

### Checklist

- [X] Bug appears at the latest library version

**Comment by d-a-bunin** [comment]
List of versions inside the image:
```
aiohttp==3.8.5
aiosignal==1.3.1
alembic==1.11.2
antlr4-python3-runtime==4.9.3
anyio==3.7.1
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
async-lru==2.0.4
async-timeout==4.0.2
attrs==23.1.0
autopage==0.5.1
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.0.0
boto3==1.28.21
botocore==1.31.21
Bottleneck==1.3.7
catboost==1.2
certifi==2023.7.22
cffi==1.15.1
charset-normalizer==3.2.0
click==8.1.6
cliff==4.3.0
cloudpickle==2.2.1
cmaes==0.10.0
cmd2==2.4.3
cmdstanpy==1.1.0
colorlog==6.7.0
comm==0.1.4
contourpy==1.1.0
convertdate==2.4.0
cycler==0.11.0
Cython==3.0.0
dask==2023.5.0
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.13
dill==0.3.7
distributed==2023.5.0
docker-pycreds==0.4.0
ephem==4.1.4
etna==2.2.0
exceptiongroup==1.1.2
executing==1.2.0
fastjsonschema==2.18.0
fonttools==4.42.0
fqdn==1.5.1
frozenlist==1.4.0
fsspec==2023.6.0
gitdb==4.0.10
GitPython==3.1.32
graphviz==0.20.1
greenlet==2.0.2
holidays==0.30
hydra-slayer==0.2.0
idna==3.4
importlib-metadata==6.8.0
importlib-resources==6.0.1
ipykernel==6.25.1
ipython==8.12.2
ipython-genutils==0.2.0
ipywidgets==7.8.0
isoduration==20.11.0
jedi==0.19.0
Jinja2==3.1.2
jmespath==1.0.1
joblib==1.3.1
json5==0.9.14
jsonpointer==2.4
jsonschema==4.19.0
jsonschema-specifications==2023.7.1
jupyter-events==0.7.0
jupyter-lsp==2.2.0
jupyter_client==8.3.0
jupyter_core==5.3.1
jupyter_server==2.7.0
jupyter_server_terminals==0.4.4
jupyterlab==4.0.4
jupyterlab-pygments==0.2.2
jupyterlab-widgets==1.1.5
jupyterlab_server==2.24.0
kiwisolver==1.4.4
lightning-utilities==0.9.0
llvmlite==0.40.1
locket==1.0.0
loguru==0.5.3
LunarCalendar==0.0.9
Mako==1.2.4
MarkupSafe==2.1.3
matplotlib==3.7.2
matplotlib-inline==0.1.6
mistune==3.0.1
msgpack==1.0.5
multidict==6.0.4
nbclient==0.8.0
nbconvert==7.7.3
nbformat==5.9.2
nest-asyncio==1.5.7
notebook==7.0.2
notebook_shim==0.2.3
numba==0.57.1
numpy==1.24.4
omegaconf==2.3.0
optuna==2.10.1
overrides==7.4.0
packaging==23.1
pandas==1.5.3
pandocfilters==1.5.0
parso==0.8.3
partd==1.4.0
pathtools==0.1.2
patsy==0.5.3
pbr==5.11.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==10.0.0
pkgutil_resolve_name==1.3.10
platformdirs==3.10.0
plotly==5.15.0
pmdarima==2.0.3
prettytable==3.8.0
prometheus-client==0.17.1
promise==2.3
prompt-toolkit==3.0.39
prophet==1.1.4
protobuf==3.20.3
psutil==5.9.5
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
Pygments==2.16.1
PyMeeus==0.5.12
pyparsing==3.0.9
pyperclip==1.8.2
python-dateutil==2.8.2
python-json-logger==2.0.7
pytorch-forecasting==0.9.2
pytorch-lightning==1.9.5
pyts==0.12.0
pytz==2023.3
PyYAML==6.0.1
pyzmq==25.1.0
referencing==0.30.2
requests==2.31.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rpds-py==0.9.2
ruptures==1.1.8
s3transfer==0.6.1
scikit-learn==1.0.2
scipy==1.10.1
seaborn==0.11.2
Send2Trash==1.8.2
sentry-sdk==1.29.2
setproctitle==1.3.2
shortuuid==1.0.11
six==1.16.0
smmap==5.0.0
sniffio==1.3.0
sortedcontainers==2.4.0
soupsieve==2.4.1
SQLAlchemy==1.4.49
stack-data==0.6.2
statsforecast==1.4.0
statsmodels==0.13.5
stevedore==5.1.0
stumpy==1.11.1
tbats==1.1.3
tblib==2.0.0
tenacity==8.2.2
terminado==0.17.1
threadpoolctl==3.2.0
tinycss2==1.2.1
toml==0.10.2
tomli==2.0.1
toolz==0.12.0
torch==1.11.0+cpu
torchmetrics==1.0.2
tornado==6.3.2
tqdm==4.65.1
traitlets==5.9.0
tsfresh==0.20.1
typer==0.4.2
types-Deprecated==1.2.9
typing_extensions==4.7.1
uri-template==1.3.0
urllib3==1.26.16
wandb==0.12.21
wcwidth==0.2.6
webcolors==1.13
webencodings==0.5.1
websocket-client==1.6.1
widgetsnbextension==3.6.5
wrapt==1.15.0
yarl==1.9.2
zict==3.0.0
zipp==3.16.2
```

**Comment by d-a-bunin** [comment]
Problem is only happening at `numpy>=1.24` and related to [removing expired deprecations](https://github.com/numpy/numpy/releases/tag/v1.24.0). The problem is fixed if `scikit-learn>=1.2.2`, but `pytorch_forecasting` has a requirement `scikit-learn<1.1,>=0.24`.

It looks like updating numpy will be possible only after solving the problems with `pytorch_forecasting`.

**Comment by martins0n** [comment]
Ok
It seems we have two ways to solve problem:
1. Limit numpy version
2. Monkeypatch numpy.warning in `etna/__init__.py`

The first one could arise conflicts during installations in some cases
The second one might complicate code execution in distributed environments  but we should test it at least


And let's install library via poetry lock in docker as you've suggested
